samples/javascript_nodejs/23.facebook-events/bots/facebookBot.js

```diff
@@ -0,0 +1,199 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+const { ActivityHandler, TurnContext, CardFactory , ActionTypes} = require('botbuilder');
+const { ChoiceFactory } = require('botbuilder-dialogs');
+// These are the options provided to the user when they message the bot
+const facebookPageIdOption = "Facebook Page Id";
+const quickRepliesOption = "Quick Replies";
+const postBackOption = "PostBack";
+
+class FacebookBot extends ActivityHandler {
+
+    /**
+     * @param {any} logger object for logging events, defaults to console if none is provided 
+     */
+    constructor(logger) {
+        super();
+        if (!logger) {
+            logger = console;
+            logger.log('[FacebookEventsBot]: logger not passed in, defaulting to console');
+        }
+
+
+        this.logger = logger;
+
+        this.onMessage(async (context) => {
+            this.logger.log('Processing a Message Activity.');
+            
+            // Show choices if the Facebook Payload from ChannelData is not handled
+            if (!await this.processFacebookPayload(context, context.activity.channelData)) {
+                if (context.activity.channelId !== "facebook") {
+                    await context.sendActivity("This sample is intended to be used with a Facebook bot.");
+                }
+                await this.showChoices(context);
+            }
+        });
+
+
+
+        this.onEvent(async (context) => {
+            this.logger.log('Processing an Event Activity.');
+
+            // Analyze Facebook payload from EventActivity.Value
+            await this.processFacebookPayload(context, context.activity.value);
+
+        });
+
+
+    }
+
+    /**
+     * 
+     * @param {TurnContext} context 
+     */
+    async showChoices(context) {
+        // Create choices for the prompt
+        const choices = [
+            {
+                value: quickRepliesOption
+            },
+            {
+                value: facebookPageIdOption
+            },
+            {
+                value: postBackOption
+            }
+        ]
+        
+        // Create the prompt message
+        var message = ChoiceFactory.forChannel(context.activity.channelId, choices, "What Facebook feature would you like to try? Here are some quick replies to choose from!");
+        await context.sendActivity(message);
+    };
+
+    /**
+     * @param {TurnContext} context 
```

Comment:
remove xml comments from method

Reply:
added this:
/**
     * Process a Facebook payload from channel data, to handle optin events, postbacks and quick replies.
     * NOTE: This is a simplification of the Facebook object model. There are many more events and payloads
     * that could be captured here. We only show some key features that are commonly used. The sample
     * can be extended to account for more Facebook-specific events according to developers' needs.
     * @param {Object} turnContext